### DISCUSSION TITLE
How to display the NodesSelection Rectangle when selecting nodes with "multiSelectionKeyCode" ?

### DISCUSSION THREAD

**Discussion by sopretty** [xyflow/xyflow#3890] [Q&A]
Hello !

[Context] 
I'm using the React flow toolbar to perform actions on selected nodes and I noticed that selecting nodes by using a "multiSelectionKeyCode" is not behaving the same way as selecting nodes by pressing Shift and creating a selection area.

[How to reproduce]

- Opens https://reactflow.dev/examples/nodes/dynamic-grouping 
- Select multiple nodes using (for mac user) "Command" + left click:
![image](https://github.com/xyflow/xyflow/assets/6479712/5c722ed4-764c-4381-a1c4-bd54a5688bce)

- Select multiple nodes using "Shift" + Drag an area containing nodes: 
![image](https://github.com/xyflow/xyflow/assets/6479712/7d67ad9b-0a33-4f63-a7d3-b613d231fd12)

I might have missed something in the documentation but do you think that there is a way for us to be able to draw that rectangle even when we select nodes one by one by using our "multiSelectionKeyCode" ? 

Thank you for your help 🙏.

**Comment by MetaMmodern** [comment]
Hey there, I recently got an answer to this:
You can take a look at this [example](https://codesandbox.io/p/devbox/selection-box-forked-cwrmf8?workspaceId=ws_TVrGSPVwBat6p6caZrVRYw) on how to do this.

(By Usman from [discord](https://discord.com/channels/771389069270712320/1449481772323835965))
